src/client/js/app.js

```diff
@@ -52,7 +53,8 @@ function startGame(type) {
 
 // Checks if the nick chosen contains valid alphanumeric characters (and underscores).
 function validNick() {
-    var regex = /^\w*$/;
+    // var regex = /^\w*$/;
+    var regex = /([A-Za-z0-9%])+/;
```

Comment:
This regex will allow a-z, A-Z (caps), 0-9, and the "%" symbol, since encodeURI will convert spaces to unicode %20